src/formatter/html.ts

```diff
@@ -0,0 +1,286 @@
+import {LintResult, LintMessage} from "../linter/LinterContext.js";
+import {LintMessageSeverity} from "../linter/messages.js";
+
+export class Html {
+	format(lintResults: LintResult[], showDetails: boolean, version: string, autofix: boolean, quiet = false): string {
+		let totalErrorCount = 0;
+		let totalWarningCount = 0;
+		let totalFatalErrorCount = 0;
+
+		// Build the HTML content
+		let resultsHtml = "";
+		lintResults.forEach(({filePath, messages, errorCount, warningCount, fatalErrorCount}) => {
+			if (!errorCount && !warningCount) {
+				// Skip files without errors or warnings
+				return;
+			}
+			// Accumulate totals
+			totalErrorCount += errorCount;
+			totalWarningCount += warningCount;
+			totalFatalErrorCount += fatalErrorCount;
+
+			// Add the file path as a section header
+			resultsHtml += `<div class="file">
+				<h3>${filePath}</h3>
+				<table>
+					<thead>
+						<tr>
+							<th>Severity</th>
+							<th>Rule</th>
+							<th>Location</th>
+							<th>Message</th>
+							${showDetails ? "<th>Details</th>" : ""}
+						</tr>
+					</thead>
+					<tbody>`;
+
+			// Sort messages by severity (fatal errors first, then errors, then warnings)
+			messages.sort((a, b) => {
+				// Handle fatal errors first to push them to the bottom
+				if (a.fatal !== b.fatal) {
+					return a.fatal ? -1 : 1; // Fatal errors go to the top
+				}
+				// Then, compare by severity
+				if (a.severity !== b.severity) {
+					return b.severity - a.severity;
+				}
+				// If severity is the same, compare by line number
+				if ((a.line ?? 0) !== (b.line ?? 0)) {
+					return (a.line ?? 0) - (b.line ?? 0);
+				}
+				// If both severity and line number are the same, compare by column number
+				return (a.column ?? 0) - (b.column ?? 0);
+			});
+
+			// Format each message
+			messages.forEach((msg) => {
+				const severityClass = this.getSeverityClass(msg.severity, msg.fatal);
+				const severityText = this.formatSeverity(msg.severity, msg.fatal);
+				const location = this.formatLocation(msg.line, msg.column);
```

Comment:
Since `formatLocation()` is an one liner, we can build this string already here and remove the method.
```suggestion
				const location = `${msg.line ?? 0}:${msg.column ?? 0}`;
```